t/RequestHandler_add_list_admin.t

```diff
@@ -0,0 +1,155 @@
+
+my $test_start_date = time();
+ok (my $spindle = Sympa::Spindle::ProcessRequest->new(
+    context          => $list,
+    action           => 'add_list_admin',
+    email            => $test_user,
+    role             => 'owner',
+    gecos          =>   $test_gecos,
+    info           =>   $available_owner_options{info},
+    profile        =>   $available_owner_options{profile},
+    reception      =>   $available_owner_options{reception},
+    visibility     =>   $available_owner_options{visibility},
+    sender     =>   $test_listmaster,
+), 'Request handler object created');
+
+ok ($spindle->spin(), 'List owner addition succeeds.');
+
+my $sdm = Sympa::DatabaseManager->instance;
+
+my $sth = $sdm->do_query('SELECT * from `admin_table` WHERE `list_admin` LIKE %s and `robot_admin` LIKE %s', $sdm->quote($test_list_name), $sdm->quote($test_robot_name));
+
+my @stored_admins;
+
+while (my $row = $sth->fetchrow_hashref()) {
+    push @stored_admins, $row;
+}
+
+is(scalar keys @stored_admins, 1, 'One admin stored in database.');
+
+is($stored_admins[0]->{user_admin}, $test_user, 'The user stored in database is the one we requested.');
+
+is($stored_admins[0]->{comment_admin}, $test_gecos, 'The user stored in database has the right gecos.');
+
+foreach my $option (keys %available_owner_options) {
+    is($stored_admins[0]->{$option.'_admin'}, $available_owner_options{$option}, "Correct value set for option $option");
+}
+
+my $new_config_file_content = read_file("$pseudo_list_directory/config");
+
+my $config_update_succeeded = 0;
+my $update_date = 0;
+my $update_author = '';
+
+foreach my $line (split '\n', $new_config_file_content) {
+    if($line =~ m{\Aupdate}) {
+        $config_update_succeeded = 1;
+    }
+    if($line =~ m{date_epoch\s+(\d+)}) {
+        $update_date = $1;
+    }
+    if($line =~ m{email\s+(\S+)}) {
+        $update_author = $1;
+    }
+}
+
+ok($config_update_succeeded, 'List config file has been updated');
+cmp_ok($update_date, '>=', $test_start_date, 'Update time is consistant with the time of test.');
+is($update_author, $test_listmaster, 'The update author name is correct.');
+
+$spindle = Sympa::Spindle::ProcessRequest->new(
+    context          => $list,
+    action           => 'add_list_admin',
+    email            => $test_user,
+    role             => 'editor',
+);
+
+ok ($spindle->spin(), 'List editor addition succeeds.');
+
+$sth = $sdm->do_query('SELECT * from `admin_table` WHERE `list_admin` LIKE %s and `robot_admin` LIKE %s', $sdm->quote($test_list_name), $sdm->quote($test_robot_name));
+
+@stored_admins = ();
+
+while (my $row = $sth->fetchrow_hashref()) {
+    push @stored_admins, $row;
+}
+
+is(scalar keys @stored_admins, 2, 'Now two admins are stored in database.');
+
+is($stored_admins[0]->{user_admin}, $test_user, 'The editor stored in database is the one we requested.');
+
+rmtree $test_directory;
+
+done_testing();
```

Comment:
Malicious cases are desirable, such as illegal values for options.

Reply:
Fixed in fee59b6.

Reply:
I see. Please add `t/RequestHandler_add_list_admin.t` to `.travis.yml`.

Reply:
I did it. However, I have a small problem: the test don't complete because the test program can't create the bulk spool. It is because the Sympa process uses the Sympa user and group defined in Constants.pm. Which is OK. however, for the tests, I modified the t/stub/Sympa/Constants.pm to set the Sympa user and group to those running the test file, so that privileges are automatically correct.
It looks like Setting the library "stub" before the normal Sympa library is not enough to force Perl to load stub preferably to Sympa sources. Any advice about how to force perl to use "t/stub/Sympa/Constants.pm" instead of "src/lib/Sympa/Constants.pm" ?

Currently, I'm doing this:

```
use lib "$Bin/stub";
use lib "$Bin/../src/lib";
```

Reply:
(I'm looking for clues about it in the docs, but if something has an advice, it is more than welcome).

Reply:
OK. I DID read the doc. It works fine now... Sorry for the noise.